docs/docs/slot-validation-actions.mdx

```diff
@@ -0,0 +1,101 @@
+
+## `ValidationAction` class
+
+You can extend the `ValidationAction` in the Rasa SDK to define custom extraction and / or validation of slots that
+can be set or updated outside of a form context.
+With this option, you do not need to specify the `action` key in the [custom slot mapping](./domain.mdx#custom-slot-mappings).
+
+Note that you do not need to implement the `name` method in the custom action extending `ValidationAction`, since this
+is already provided: `action_validate_slot_mappings`. However this name must be added to the domain `actions` list.
+
+:::note
+Slots mapped to a form in the mapping `conditions` will be ignored by `ValidationAction` unless you override
+the `required_slots` method to include the slots you want to extract and / or validate.
+:::
+
+### Validation of Slots with Predefined Mappings
+
+You must write functions named `validate_<slot_name>` for every predefined slot that you need validated.
+
+In the following example, the value for slot `location` is capitalized only if the extracted value is of type string:
+
+```python
+from typing import Text, Any, Dict
+
+from rasa_sdk import Tracker, ValidationAction
+from rasa_sdk.executor import CollectingDispatcher
+from rasa_sdk.types import DomainDict
+
+
+class ValidatePredefinedSlots(ValidationAction):
+    def validate_location(
+        self,
+        slot_value: Any,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: DomainDict,
+    ) -> Dict[Text, Any]:
+        """Validate location value."""
+        if isinstance(slot_value, str):
+            # validation succeeded, capitalize the value of the "location" slot
+            return {"location": slot_value.capitalize()}
+        else:
+            # validation failed, set this slot to None
+            return {"location": None}
+```
+
+### Extraction of Custom Slot Mappings
+
+In order to define custom extraction code, you must write an `extract_<slot_name>` method for every slot with a
+custom slot mapping.
+
+The following example shows the implementation of a custom action that extracts the slot `count_of_insults` to keep
+track of the user's attitude.
+
+```python
+from typing import Dict, Text, Any
+
+from rasa_sdk import Tracker
+from rasa_sdk.executor import CollectingDispatcher
+from rasa_sdk.forms import ValidationAction
+
+
+class ValidateCustomSlotMappings(ValidationAction):
+    async def extract_count_of_insults(
+        self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict
+    ) -> Dict[Text, Any]:
+        intent_of_last_user_message = tracker.get_intent_of_latest_message()
+        current_count_of_insults = tracker.get_slot("count_of_insults")
+        if intent_of_last_user_message == "insult":
+           current_count_of_insults += 1
+
+        return {"count_of_insults": current_count_of_insults}
+```
+
+## `FormValidationAction` class
+
+If you prefer that certain custom slot mappings should only be set within the context of a form,
```

Comment:
Bit confused by this condition phrasing because `conditions` already allow you to limit applicability of any mapping to a form. 
Will all validation actions run regardless of conditions? If so, this could read:
```suggestion
If certain custom slot mappings should only be validated within the context of a form,
```

Reply:
The mapping `conditions` actually have no effect if a custom extraction was defined in an action extending `ValidationAction` instead of `FormValidationAction`. That's why I tried to emphasise that if there are any custom slot mappings meant to be both extracted and validated only within a form, better use `FormValidationAction`.